### ISSUE TITLE
Number Range

### ISSUE THREAD

**Issue by ktrianta** [souffle-lang/souffle#388]
In some cases number's fixed range ([-2^31 .. 2^31-1]) leads to integer overflow (e.g., counting the contents of a large relation) or inability to parse a numeric constant that is out of range. It is not clear to me how to change the type from int32_t to int64_t as the manual describes

> the type could be altered by a compile time flags to any basic type that the C++ compiler supports

Specifically, I built Soufflé after running
```CXXFLAGS="-DRAM_DOMAIN_TYPE=int64_t" ./configure```
but it didn't change something except when I use ` souffle -c` I get
```In file included from /home/triantaf/souffle/bin/../include/souffle/CompiledSouffle.h:21:0,
                 from ./souffleW6kppW.cpp:1:
/home/triantaf/souffle/bin/../include/souffle/CompiledRamRecord.h:80:45: error: right operand of shift expression ‘(1 << 44ul)’ is >= than the precision of the left operand [-fpermissive]
     static const std::size_t NUM_BLOCKS = 1 << (sizeof(RamDomain) * 8 - 20);
                                           ~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/triantaf/souffle/bin/../include/souffle/CompiledRamRecord.h:95:5: error: ‘block_index_type’ does not name a type
     block_index_type i2r;
     ^~~~~~~~~~~~~~~~
/home/triantaf/souffle/bin/../include/souffle/CompiledRamRecord.h: In member function ‘souffle::RamDomain souffle::detail::RecordMap<Tuple>::pack(const tuple_type&)’:
/home/triantaf/souffle/bin/../include/souffle/CompiledRamRecord.h:128:30: error: ‘i2r’ was not declared in this scope
                 auto& list = i2r[index / BLOCK_SIZE];
                              ^~~
/home/triantaf/souffle/bin/../include/souffle/CompiledRamRecord.h: In member function ‘const tuple_type& souffle::detail::RecordMap<Tuple>::unpack(souffle::RamDomain)’:
/home/triantaf/souffle/bin/../include/souffle/CompiledRamRecord.h:145:19: error: ‘i2r’ was not declared in this scope
         return (*(i2r[index / BLOCK_SIZE]))[index % BLOCK_SIZE];
```

**Comment by b-scholz** [comment]
Yes - indeed this is an issue. Unfortunately there is no quick fix. Ideally, we change the type system. The first change would be manageable, i.e., introducing signed, unsigned integer and float types of the  same size (either 32bit or 64bit). Having different type sizes would be more involved.

**Comment by mmcgr** [comment]
Fixed in #569 . #579 adds a configure script flag `--enable-64bit-domain` to enable 64 bit domains. (Note that this just fixes support of 64bit RamDomains - integer overflow is still possible, this just increases the range.)
